src/poprox_storage/repositories/experiments.py

```diff
@@ -148,6 +122,77 @@ def get_active_expt_allocations(self, date: datetime.date | None = None) -> dict
 
         return group_lookup_by_account
 
+    def _insert_experiment(self, experiment: Experiment) -> UUID | None:
+        return self._insert_model("experiments", experiment, exclude={"phases"}, commit=False)
+
+    def _insert_expt_group(
+        self,
+        experiment_id: UUID,
+        group: Group,
+    ) -> UUID | None:
+        return self._insert_model(
+            "expt_groups", group, {"experiment_id": experiment_id}, exclude={"minimum_size"}, commit=False
+        )
+
+    def _insert_expt_recommender(
+        self,
+        experiment_id: UUID,
+        recommender: Recommender,
+    ) -> UUID | None:
+        return self._insert_model(
+            "expt_recommenders",
+            recommender,
+            {
+                "recommender_name": recommender.name,
+                "experiment_id": experiment_id,
+            },
+            exclude={"name"},
+            commit=False,
+        )
+
+    def _insert_expt_phase(
+        self,
+        experiment_id: UUID,
+        phase: Phase,
+    ) -> UUID | None:
+        return self._insert_model(
+            "expt_phases",
+            phase,
+            {"phase_name": phase.name, "experiment_id": experiment_id},
+            exclude={"name", "treatments"},
+            commit=False,
+        )
+
+    def _insert_expt_treatment(
+        self,
+        phase_id: UUID,
+        treatment: Treatment,
+    ) -> UUID | None:
+        return self._insert_model(
+            "expt_treatments",
+            treatment,
+            {
+                "phase_id": phase_id,
+                "group_id": treatment.group.group_id,
+                "recommender_id": treatment.recommender.recommender_id,
+            },
+            exclude={"group", "recommender"},
+            commit=False,
+        )
+
+    def _insert_expt_assignment(
```

Comment:
This method follows the pattern of the methods above that each take a Pydantic model and a foreign key id, but I cheated here and passed a `Group` object that's only used to grab the group id. This code was written before the `Allocation` object had been created, but it should probably be used here.